src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -98,12 +101,13 @@ public TextSubmission handleTextSubmission(TextSubmission textSubmission, TextEx
      * @return the textSubmission entity that was saved to the database
      */
     @Transactional(rollbackFor = Exception.class)
-    public TextSubmission save(TextSubmission textSubmission, StudentParticipation participation) {
+    public TextSubmission save(TextSubmission textSubmission, StudentParticipation participation, Principal principal) {
         // update submission properties
         textSubmission.setSubmissionDate(ZonedDateTime.now());
         textSubmission.setType(SubmissionType.MANUAL);
         textSubmission.setParticipation(participation);
         textSubmission = textSubmissionRepository.save(textSubmission);
+        submissionVersionService.save(textSubmission, principal.getName());
```

Comment:
can we wrap this in a generic try {... } catch(Exception ex) {...} so that it never can happen that any kind of (runtime) exception (we did not think of) in the submission version service makes problem with saving the normal submission? (which should really always work!)

Here, we might also make sure the transaction is not rolled back if an exception in submissionVersionService occurs

Reply:
Will do.